engine_graphics/include/core/gl_extensions.hpp

```diff
@@ -0,0 +1,848 @@
+///////////////////////////////////////////////////////////////////////////////
+// glExtension.h
+// =============
+// OpenGL extension helper
+// If an ARB extension is included in OpenGL core, it uses the function poiters
+// and tokens without ARB suffix.
+// NOTE: In order to get valid OpenGL extensions on Windows system, HDC must be
+// passed to wglGetExtensionsStringARB() function. The size of HDC in 64bit
+// Windows is 8 bytes.
+//
+// extensions
+// ==========
+// GL_ARB_framebuffer_object
+// GL_ARB_debug_output
+// GL_ARB_direct_state_access
+// GL_ARB_multisample
+// GL_ARB_multitexture
+// GL_ARB_pixel_buffer_objects, GL_ARB_vertex_buffer_object
+// GL_ARB_shader_objects, GL_ARB_vertex_program, GL_ARB_fragment_program, GL_ARB_vertex_shader, GL_ARB_fragment_shader
+// GL_ARB_sync
+// GL_ARB_vertex_array_object
+// WGL_ARB_extensions_string
+// WGL_ARB_pixel_format
+// WGL_ARB_create_context
+// WGL_EXT_swap_control
+//
+//  AUTHOR: Song Ho Ahn (song.ahn@gmail.com)
+// CREATED: 2013-03-05
+// UPDATED: 2018-08-09
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef GL_EXTENSION_H
+#define GL_EXTENSION_H
+
+// in order to get function prototypes from glext.h, define GL_GLEXT_PROTOTYPES before including glext.h
+#define GL_GLEXT_PROTOTYPES
+
+#ifdef _WIN32
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN 1
+#endif
+#include <windows.h>
+#endif
+
+#ifdef __APPLE__
+#include <OpenGL/gl.h>
+#else
+#include <GL/gl.h>
+#endif
+
+#include <string>
+#include <vector>
+//#include <GL/glew.h>
+
+class glExtension
+{
+public:
+    ~glExtension();
+    static glExtension& getInstance();                  // must be called after RC is open
+
+    bool isSupported(const std::string& extStr);        // check if a extension is available
+    const std::vector<std::string>& getExtensions();
+
+private:
+    glExtension();                                      // prevent calling ctor
+    glExtension(const glExtension& rhs);                // no implementation
+    void getExtensionStrings();
+    void getFunctionPointers();
+    std::string toLower(const std::string& str);
+
+    std::vector <std::string> extensions;
+};
+
+
+
+#ifdef _WIN32 //===============================================================
```

Comment:
Remove windows related stuff.

Reply:
We should keep this engine as cross-platform, as possible, The plan is to actually make it cross-platform in the future, so this should be left alone.